helm-chart/renku/templates/graph-token-postgres-secret.yaml

```diff
@@ -15,6 +24,11 @@ metadata:
     "helm.sh/hook": "pre-install,pre-rollback"
     {{- end }}
     "helm.sh/hook-delete-policy": "before-hook-creation"
+    {{ if .Values.global.externalServices.postgresql.enabled }}
+    "helm.sh/resource-policy": "keep"
+    {{- else -}}
+    "helm.sh/hook-delete-policy": "before-hook-creation"
```

Comment:
We need keep if we use an external Postgres to stop the password value from being changed on the reinstallation of Renku

Reply:
@wesjdj here are the annotations in the file we are discussing:
```
  annotations:
    {{ if or .Values.global.graph.tokenRepository.postgresPassword.value .Values.global.graph.tokenRepository.postgresPassword.overwriteOnHelmUpgrade -}}
    "helm.sh/hook": "pre-install,pre-upgrade,pre-rollback"
    {{- else -}}
    "helm.sh/hook": "pre-install,pre-rollback"
    {{- end }}
    "helm.sh/hook-delete-policy": "before-hook-creation" <<<<<<<<<< This will always be templated no matter what
    {{ if .Values.global.externalServices.postgresql.enabled }}
    "helm.sh/resource-policy": "keep"
    {{- else -}}
    "helm.sh/hook-delete-policy": "before-hook-creation"  <<<<<<<<< 
    {{- end }}
```

The line `"helm.sh/hook-delete-policy": "before-hook-creation"` shows up twice. Once outside of any if/else statement and then again below in the `else` clause of an if/else statement.

This second part is redundant because the part outside of the if/else statement will always be templated. So we should either put the first part in an if/else statement or remove the whole else statement that contains the second repetition of the line. I am still muddy on the logic and meaning of all the hooks to know whether this should always be templated or it should be only templated in some cases. So I am not sure if the else clause should be removed or the bare statement that will always be templated.

This occurs in several other files.

Reply:
Removed the redundant "helm.sh/hook-delete-policy": "before-hook-creation"